bin/XMLToSARIF.sh

```diff
@@ -0,0 +1,125 @@
+print_usage() {
+    echo "Usage: XMLToSARIF.sh --inputXmlReport  <inputXmlReport> [--outputSarifReport <outputSarifReport>] [--javaOrToolHomeDir <javaOrToolHomeDir>]"
+    echo ""
+    echo "Options:"
+    echo "  --inputXmlReport      (required) Path to the input XML report."
+    echo "  --outputSarifReport   Path to the output SARIF report."
+    echo "  --javaOrToolHomeDir         Path to the tool home directory."
+    echo ""
+}
+
+check_param() {
+    # check if the parameter is empty or starts with "--"
+    local param_name="$1"
+    local param_value="$2"
+    if [[ -z "$param_value" || "$param_value" == --* ]]; then
+        echo "Error: Empty value passed to option $param_name "
+        exit 1
+    fi
+}
+
+getJavaPath() {
+    local javaOrParasoftToolRootPath="$1"
+
+    if [ ! -d "$javaOrParasoftToolRootPath" ]; then
+        echo "$javaOrParasoftToolRootPath is not found."
+        exit 1;
+    fi
+
+    local javaFileName="java"
+    # bin -- Find java executable in Java home directory
+    # bin/dottest/Jre_x64/bin -- Find java executable in DotTest installation directory
+    # bin/jre/bin -- Find java executable in Jtest,C++Test installation directory
+    local javaPaths=("bin" "bin/dottest/Jre_x64/bin" "bin/jre/bin")
+
+    for path in "${javaPaths[@]}"; do
+        local javaFilePath="$javaOrParasoftToolRootPath/$path/$javaFileName"
+        if [ -f "$javaFilePath" ]; then
+            echo "$javaFilePath"
+            return 0
+        fi
+    done
+
+    return 1
+}
+
+# 1.Initialize variables:
+# Base directory of the transformer
+BASE_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
+SARIF_XSL_PATH="$BASE_DIR/xsl/sarif.xsl"
+
+java_or_parasoft_tool_root_path=""
+xml_report_path=""
+sarif_report_path=""
+
+# 2. Process options into variables.
+while [[ $# -gt 0 ]] ; do
+  case "$1" in
+    --inputXmlReport)
+        xml_report_path="$2"
+        check_param "$1" "$xml_report_path"
+        shift 2 ;;
+    --outputSarifReport)
+        sarif_report_path="$2"
+        check_param "$1" "$sarif_report_path"
+        shift 2 ;;
+    --javaOrToolHomeDir)
+        java_or_parasoft_tool_root_path="$2"
+        check_param "$1" "$java_or_parasoft_tool_root_path"
+        shift 2 ;;
+    --) shift ; break ;;
+    *)
+        echo "Error: Invalid option $1"
+        print_usage
+        exit 1 ;;
+  esac
+done
+
+# 3. Validate options value
+
+# Validate options value
+if [ -z "$xml_report_path" ]; then
+    echo "Error: inputXmlReport is required."
+    print_usage
+    exit 1
+elif [ ! -f "$xml_report_path" ]; then
+    echo "Error: inputXmlReport file not found."
+    exit 1
+fi
+
+# Set default SARIF report path if not provided
+if [ -z "$sarif_report_path" ]; then
```

Comment:
I think we can just keep the `if` condition and get rid of variable `sarif_report`